server/src/main/java/org/apache/iotdb/db/metadata/mtree/MTree.java

```diff
@@ -1011,39 +1078,51 @@ protected void transferToResult(IMNode node) {
   /**
    * Get the count of timeseries matching the given path.
    *
-   * @param path a path pattern or a full path, may contain wildcard
+   * @param pathPattern a path pattern or a full path, may contain wildcard
    */
-  public int getAllTimeseriesCount(PartialPath path) throws MetadataException {
-    CounterTraverser counter = new MeasurementCounter(root, path);
+  public int getAllTimeseriesCount(PartialPath pathPattern) throws MetadataException {
+    CounterTraverser counter = new MeasurementCounter(root, pathPattern);
+    counter.traverse();
+    return counter.getCount();
+  }
+
+  /**
+   * Get the count of timeseries component matching the given path.
+   *
+   * @param pathPattern a path pattern or a full path, may contain wildcard
+   */
+  public int getAllTimeseriesFlatCount(PartialPath pathPattern) throws MetadataException {
+    CounterTraverser counter = new MeasurementFlatCounter(root, pathPattern);
     counter.traverse();
```

Comment:
put the traverse to getCount() methods looks more compact;

getCount() {
if(!hasTraverse) {
traverse;
hasTraverse=true;
}
return xxx;
}

Reply:
I think keep two these two steps is clear and flexible. It helps remind user to invoke traverse() before getCount() while the MTree is changed.